libpermian/workflows/test_workflows.py

```diff
@@ -44,3 +45,60 @@ def test_unknown(self, MockTestRuns):
         testRuns.settings = Settings({}, {}, [])
         WorkflowFactory._assignWorkflows('unknown', testRuns, testRuns.caseRunConfigurations)
         self.assertIsInstance(caseRunConfiguration.workflow, UnknownWorkflow)
+
+
+class SilentlyFailingWorklfow(GroupedWorkflow):
+    silent_exceptions = (NotImplementedError,)
+    def __init__(self, testRuns, crcList):
+        super().__init__(testRuns, crcList)
+        self.mocked_log = ''
+
+    def setup(self):
+        raise NotImplementedError('This is currently not supported')
+
+    def groupLog(self, text, **kwargs):
+        self.mocked_log += text
+
+    def execute(self):
+        pass
+    def factory(self):
+        pass
+    def groupDisplayStatus(self):
+        pass
+    def groupTerminate(self):
+        pass
+
+class BrokenWorklfow(GroupedWorkflow):
+    def execute(self):
+        raise ZeroDivisionError('I\'m broken')
+    def groupReportResult(self, crclist, result):
+        pass
+    def factory(self):
+        pass
+    def groupDisplayStatus(self):
+        pass
+    def groupTerminate(self):
+        pass
+
+class TestWorkflowExceptions(unittest.TestCase):
+    @unittest.mock.patch('libpermian.testruns.TestRuns', autospec=True)
+    def setUp(self, MockTestRuns):
+        self.crc = CaseRunConfiguration(DummyTestCase(), {}, [])
+        self.mock_testrun = MockTestRuns(None, None, None)
+        self.mock_testrun.caseRunConfigurations = CaseRunConfigurationsList([self.crc])
+        self.mock_testrun.event = None
+        self.mock_testrun.settings = Settings({}, {}, [])
+
+    def test_silent_exception(self):
+        workflow = SilentlyFailingWorklfow(self.mock_testrun, [self.crc])
+        workflow.run()
+        self.assertEqual(workflow.mocked_log, 'Workflow failed with silent exception: This is currently not supported')
+
+    def test_exception(self):
+        workflow = BrokenWorklfow(self.mock_testrun, [self.crc])
+        try:
```

Comment:
I believe that https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises does exactly the thing needed here.